packages/ui/src/validators/hooks/useValidatorMembers.tsx

```diff
@@ -0,0 +1,38 @@
+import { useMemo } from 'react'
+import { map } from 'rxjs'
+
+import { useApi } from '@/api/hooks/useApi'
+import { useFirstObservableValue } from '@/common/hooks/useFirstObservableValue'
+import { useGetMembersWithDetailsQuery } from '@/memberships/queries'
+import { asMemberWithDetails } from '@/memberships/types'
+
+import { ValidatorMembership } from '../types'
+
+export const useValidatorMembers = () => {
+  const { api } = useApi()
+  const allValidatorAddresses = useFirstObservableValue(
+    () =>
+      api?.query.staking.validators
+        .entries()
+        .pipe(map((entries) => entries.map((entry) => entry[0].args[0].toString()))),
+    [api?.isConnected]
+  )
+
+  const variables = {
+    where: { metadata: { validatorAccount_in: allValidatorAddresses } },
```

Comment:
Same thing here. Memberships shouldn't be fetch based on the `validatorAccount` field.

Instead:
1. Each validator (stashing) account is associated to a validator controller account. This mapping should be available via an RPC node query.
2. The membership profile to show for a given validator is the one that is bound to either its validator stashing account or its validator controller account.
 
FYI about (2) ATM Pioneer will be strongly suggesting to bind validator controller accounts but controller accounts will eventually get deprecated in favor of [Proxy Accounts](https://wiki.polkadot.network/docs/learn-proxies). By then validators will be binding there stashing accounts via proxies. This is why both type binding should be supported.